src/hash.rs

```diff
@@ -64,3 +70,65 @@ fn file_hash(algorithm: &str, path: &str) -> Result<String> {
 
     Ok(hash_algorithm(algorithm, &bytes)?)
 }
+
+fn totp_generate(hex_seed: &str) -> Result<String> {
+    let mut seed: [u8; 64] = [0; 64];
+
+    let hex_seed_bytes = (0..hex_seed.len())
+                        .step_by(2)
+                        .map(|i| u8::from_str_radix(&hex_seed[i..i + 2], 16));
+
+    let mut i = 0;
+    for b in hex_seed_bytes {
+        seed[i] = b.unwrap();
+        i += 1;
+    }
+
+    let mut ipad: [u8; 64] = [0; 64];
+    let mut opad: [u8; 64] = [0; 64];
+
+    for j in 0..64 {
+        ipad[j] = seed[j] ^ 0x36;
+        opad[j] = seed[j] ^ 0x5C;
+    }
+    let time: u64 = SystemTime::now().duration_since(UNIX_EPOCH).expect("SystemTime before UNIX EPOC").as_secs() / 30;
+
+    let time_bytes: [u8; 8] = time.to_be_bytes();
+
+    let mut ipad_time: [u8; 72] = [0; 72];
+
+    for j in 0..72 {
+        if j < 64 {
+            ipad_time[j] = ipad[j];
+        } else {
+            ipad_time[j] = time_bytes[j - 64];
+        }
+    }
+
+    let mut hasher = Sha1::new();
+    hasher.update(ipad_time);
+    let ipad_time_hash = hasher.finalize();
+
+    let mut ipad_time_hash_opad: [u8; 84] = [0; 84];
+
+    for j in 0..84 {
+        if j < 64 {
+            ipad_time_hash_opad[j] = opad[j]
+        } else {
+            ipad_time_hash_opad[j] = ipad_time_hash[j - 64]
+        }
+    }
```

Comment:
Same story as the rest really. I think you can just get rid of this entire array (and through doing so get rid of the loop.)